Server/htdocs/AppController/commands_RSM/api/v2/items/update.php

```diff
@@ -0,0 +1,106 @@
+<?php
+// ****************************************************************************************
+//Description:
+//  Edits one or more items of the specified itemTypeID with the associated values
+//
+//  REQUEST BODY (JSON)
+// Array with object/s inside, each object must contain
+//          - one propertyId and its value (one or more)
+//          - id: id of of the item being updated
+//  EXAMPLE:
+//      [{
+//          '109': 'Roja'
+//          'ID': '0008'
+//        },{
+//          'ID':  '4001'
+//          '319': "Peter
+//          '320': 'Parker'
+//        }]
+// ****************************************************************************************
+
+require_once '../../../utilities/RStools.php';
+require_once '../../../utilities/RSMverifyBody.php';
+setAuthorizationTokenOnGlobals();
+checkCorrectRequestMethod('PATCH');
+
+require_once '../../../utilities/RSdatabase.php';
+require_once '../../../utilities/RSMitemsManagement.php';
+require_once '../../api_headers.php';
+
+$requestBody = getRequestBody();
+verifyBodyContent($requestBody);
+
+// Definitions
+$clientID = getClientID();
+$RStoken = getRStoken();
+$RSuserID = getRSuserID();
+
+foreach ($requestBody as $item) {
+  $propertiesID = array();
+  //Iterate through every propertyID of the items to check if they pertain to the same item type
+  foreach ($item as $propertyID => $propertyValue) {
+    if ($propertyID != 'ID') {
+      $propertiesID[] = ParsePID($propertyID, $clientID);
+    }
+  }
+  $itemTypeID = getItemTypeIDFromProperties($propertiesID, $clientID);
+  foreach ($propertiesID as $propertyID) {
+    if (!(RShasTokenPermission($RStoken, $propertyID, 'WRITE') || isPropertyVisible($RSuserID, $propertyID, $clientID))) {
+      $RSallowDebug ? returnJsonMessage(400, 'Not Updated (At least 1 property has no WRITE permissions or is not visible)')  : returnJsonMessage(400, '');
+    }
+  }
+  $itemID = $item->ID;
+  if ($itemTypeID == 0) {
+    $RSallowDebug ? returnJsonMessage(400, 'Not Updated (Properties must pertain to the same item type)') : returnJsonMessage(400, '');
+  } elseif (!verifyItemExists($itemID, $itemTypeID, $clientID)) {
+    $RSallowDebug ? returnJsonMessage(400, 'Item doesn\'t exist') : returnJsonMessage(400, '');
+  }
+}
+
+foreach ($requestBody as $item) {
+  $itemTypeID = getItemTypeIDFromProperties($propertiesID, $clientID);
```

Comment:
$propertiesID should be $propertyIDs. And you can't use that variable here, because the value pertains to the latest iteration of the previous loop. $propertyIDs should be obtained in each iteration.